OSAT-VideoCompositor/Classes/OSATVideoComposition.swift

```diff
@@ -128,4 +128,152 @@ public struct OSATVideoComposition {
+              let compositionAudioTrack = composition.addMutableTrack(withMediaType: .audio, preferredTrackID: kCMPersistentTrackID_Invalid)  else {
+            NSLog("track not created.", "")
+            errorHandler(.mutableTrackFailed)
+            return
+        }
+        
+        // build composition
+        for sourceVideoUrl in sourceVideoURL {
+            let asset = AVAsset(url: sourceVideoUrl)
+            if asset.tracks(withMediaType: .video).count != 0 {
+                do {
+                    try compositionVideoTrack.insertTimeRange(CMTimeRange(start: .zero, duration: asset.duration), of: asset.tracks(withMediaType: .video)[0], at: nextClipStartTime)
+                } catch { print(error) }
+            }
+            
+            if asset.tracks(withMediaType: .audio).count != 0 {
+                do {
+                    try compositionAudioTrack.insertTimeRange(CMTimeRange(start: .zero, duration: asset.duration), of: asset.tracks(withMediaType: .video)[0], at: nextClipStartTime)
+                } catch { print(error) }
+            }
+            // build transition instruction
+            let instruction = AVMutableVideoCompositionInstruction()
+            instruction.timeRange = CMTimeRange(start: nextClipStartTime, duration: asset.duration)
+            let layerInstruction = compositionLayerInstruction(for: compositionVideoTrack, assetTrack: asset.tracks(withMediaType: .video)[0])
+            instruction.layerInstructions = [layerInstruction]
+            videoInstructions.append(instruction)
+            nextClipStartTime = CMTimeAdd(nextClipStartTime, asset.duration)
+        }
+        let asset = AVAsset(url: sourceVideoURL[0])
+        let assetTrack = asset.tracks(withMediaType: .video)[0]
+        
+        let videoInfo = orientation(from: assetTrack.preferredTransform)
+        let videoSize: CGSize = videoInfo.isPortrait ? CGSize(width: assetTrack.naturalSize.height, height: assetTrack.naturalSize.width) : assetTrack.naturalSize
+        
+        compositionVideoTrack.preferredTransform = assetTrack.preferredTransform
+        
+        let videoComposition = AVMutableVideoComposition()
+        videoComposition.renderSize = videoSize
+        videoComposition.frameDuration = CMTime(value: 1, timescale: 30)
+        videoComposition.instructions = videoInstructions
+        
+        guard let export = AVAssetExportSession(
+            asset: composition,
+            presetName: AVAssetExportPresetHighestQuality)
+        else {
+            NSLog("Cannot create export session.", "")
+            errorHandler(.assetExportSessionFailed)
+            return
+        }
+        
+        let videoName = UUID().uuidString
+        let exportURL = exportURL.appendingPathComponent(videoName).appendingPathExtension("mov")
+        
+        export.videoComposition = videoComposition
```

Comment:
Can we make a separate function for exporting video composition? 

 let videoName = UUID().uuidString
        let exportURL = exportURL.appendingPathComponent(videoName).appendingPathExtension("mov")

        export.videoComposition = videoComposition
        export.outputFileType = .mov
        export.outputURL = exportURL

        export.exportAsynchronously {
            DispatchQueue.main.async {
                switch export.status {
                case .completed:
                    completionHandler(export)
                default:
                    errorHandler(.assetExportSessionFailed)
                    break
                }
            }
        }